### PR TITLE
[3D] Unify some some shader code

### PR DESCRIPTION
**Pull request by ptitjano**
## Description

This PR is factored out from https://github.com/qgis/QGIS/pull/57899 
It contains some unification for the shader code:
- Use the default vertex code for `QgsPhongTexturedMaterialSettings`
- Use the default vertex code for `QgsPhongMaterialSettings`
- Merge the constant and data defined cases of `QgsPhongMaterialSettings` with one shader file by using a `#ifdef` logic
- Merge the constant and data defined cases of `QgsGoochMaterialSettings` with one shader file by using a `#ifdef` logic

### CODE REVIEW

**Review by benoitdm-oslandia — APPROVED** [review]
LGTM!

File: src/3d/shaders.qrc [hunk 1]
```diff
@@ -24,17 +23,15 @@
         <file>shaders/preview.vert</file>
         <file>shaders/preview.frag</file>
         <file>shaders/phongDataDefined.vert</file>
-        <file>shaders/phongDataDefined.frag</file>
-        <file>shaders/goochDataDefined.frag</file>
```
Comment (nyalldawson) [inline src/3d/shaders.qrc:28]: Looks like this file needs deleting too

File: src/3d/materials/qgsphongtexturedmaterialsettings.cpp [hunk 2]
```diff
@@ -141,7 +141,7 @@ Qt3DRender::QMaterial *QgsPhongTexturedMaterialSettings::toMaterial( QgsMaterial
       Qt3DRender::QShaderProgram *shaderProgram = new Qt3DRender::QShaderProgram();
 
       //Load shader programs
-      const QUrl urlVert( QStringLiteral( "qrc:/shaders/diffuseSpecular.vert" ) );
```
Comment (nyalldawson) [inline src/3d/materials/qgsphongtexturedmaterialsettings.cpp:144]: diffuseSpecular.vert should be deleted and removed from shaders.qrc

Comment (ptitjano) [reply src/3d/materials/qgsphongtexturedmaterialsettings.cpp:144]: Fixed. This was introduced during a rebase.

File: src/3d/shaders.qrc (hunk 1 above)
Comment (nyalldawson) [reply src/3d/shaders.qrc:28]: Ah, missed that there's a rename here -- I was looking for the removal of the goochDataDefined file

**Review by nyalldawson — APPROVED** [review]

**Comment by wonder-sk** [comment]
good stuff! :clap:

**Pull request merged**